actionpack/lib/action_dispatch/middleware/ssl.rb

```diff
@@ -114,25 +114,33 @@ def build_hsts_header(hsts)
       end
 
       def flag_cookies_as_secure!(headers)
-        if cookies = headers["Set-Cookie"]
-          if cookies.is_a?(String)
-            cookies = cookies.split("\n")
-          end
+        cookies = headers[Rack::SET_COOKIE]
+        return unless cookies
 
-          headers["Set-Cookie"] = cookies.map { |cookie|
+        if Gem::Version.new(Rack::RELEASE) < Gem::Version.new("3")
+          cookies = cookies.split("\n") if cookies.is_a?(String)
```

Comment:
Should we unconditionally split here? I think in Rack 2 all headers must be strings.

Reply:
Yes good call!